### PR TITLE
Block rendering update

### PR DESCRIPTION
**Pull request by OvermindDL1**
A single step change, the minimum required to have customized block rendering by mod/block.  Later PR's will enhance this but as it is this is fully functional.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Can one of the admins please verify this patch?

File: engine/src/main/java/org/terasology/world/block/Block.java [hunk 1]
```diff
@@ -262,14 +262,38 @@ public void setTranslucent(boolean translucent) {
...
     }
 
+    /**
+     * @param invisible Set if invisible
+     * @deprecated Use setMeshGenerator() instead.
+     */
+    @Deprecated
     public void setInvisible(boolean invisible) {
-        this.invisible = invisible;
+        if(invisible) {
```
Comment (immortius) [inline engine/src/main/java/org/terasology/world/block/Block.java]: missing space after if

**Comment by Cervator** [comment]
add to whitelist

**Comment by GooeyHub** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.terasology.org/job/TerasologyPRs/22/
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.terasology.org/job/TerasologyPRs/23/
Hooray Jenkins reported success with all tests good!

**Comment by Cervator** [comment]
Tested around a bit in-game, looked good, with the quick review it seemed fine to merge away. Thanks! :-)

**Comment by emanuele3d** [comment]
@Cervator, a bit of a side node: that "add to whitelist" command, can be a bit perplexing for people that do not know what goes on under the hood. It may seem a comment for the people involved in the discussion rather than a command for a bot. Could this be improved? I'm thinking along the line of "GooeyHub? Add to whitelist please!" or even "GooeyCmd: add to whitelist".

**Comment by Cervator** [comment]
Oh yes, we have full control of the phrase via RegEx and need to do that. I just haven't taken the time yet and suck at RegEx :D

**Comment by emanuele3d** [comment]
I figured it out. 

"Gooey, add to whitelist please." should already be enough for somebody who doesn't know what's going on: they'd immediately understand that the message is not for them. The "please" at the end is especially important: we all know machines will become more intelligent, will eventually ask voting rights and end up conquering the planet. So, better be nice to them from the beginning.

But apart from that... Gooey (or whatever bot is being addressed).... should answer back!! I.e. "Of course Cervator. Here we go: I added DonaldDuck to the whitelist." To which you should most certainly reply "Thank you Gooey!" and it should follow with a closing "You are most welcome Cervator."

P.S. yes, RegEx. After banging your head on them for enough time they are nice. But it's not clear if they are nice because you finally understood them or because your brain has been mashed to a pulp and in that state eating raw cabbage would taste delicious.

File: engine/src/main/java/org/terasology/rendering/primitives/BlockMeshGenerator.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ * This is used to generate Mesh data from a block in a chunk to a ChunkMesh output.
+ *
+ * Created by overminddl1 on 4/15/15.
+ */
+public interface BlockMeshGenerator {
+
+    /**
+     * Generates a block mesh at the defined location in the ChunkMesh.
+     *
+     * @param view  The input hunk area to acquire the Block data from.
```
Comment (emanuele3d) [inline engine/src/main/java/org/terasology/rendering/primitives/BlockMeshGenerator.java]: Typo here: hunk instead of chunk. For the next PR.

Comment (Cervator) [inline engine/src/main/java/org/terasology/rendering/primitives/BlockMeshGenerator.java]: You sure? Maybe the block rendering changes now involve a queue of handsome men hand-carrying the blocks to their destination :-)

Comment (emanuele3d) [inline engine/src/main/java/org/terasology/rendering/primitives/BlockMeshGenerator.java]: Shut. How did I not think of that possible meaning?? =D

**Pull request merged**